### PR TITLE
PUD-1007: Store prisoner first, middle & last names, and a category to define how the name should be formatted

### PR DESCRIPTION
**Pull request by philsegal**
- utilise the appropriate format of prisonerFullName on all documents.

### CODE REVIEW

File: localstack-docker/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,3 @@
-FROM localstack/localstack
+FROM localstack/localstack:0.12.20
```
Comment (philsegal) [inline localstack-docker/Dockerfile:1]: 0.13.0 doesnt load the api_recorded_calls.json and so doesn't create the bucket. Need to find how to fix it.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Recall.kt [hunk 2]
```diff
@@ -199,6 +226,26 @@ data class Recall(
 
   fun recallAssessmentDueDateTime(): OffsetDateTime? = recallEmailReceivedDateTime?.plusHours(24)
 
+  fun prisonerFullName(): FullName =
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Recall.kt:229]: A bewildering array of types and methods for combinations of person name properties ... 
Just from the code this looks more like 
`fun prisonerNameOnLicense(): FullName` 
than 
`fun prisonerFullName(): FullName`?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/DocumentFormatters.kt [hunk 3]
```diff
@@ -79,9 +78,8 @@ fun MappaLevel.shouldShowOnDocuments(): Boolean {
   }
 }
 
-fun UserDetails.personName(): String = PersonName(this.firstName, lastName = this.lastName).firstAndLastName()
-fun Prisoner.personName(): PersonName =
-  PersonName(this.firstName!!, this.middleNames, this.lastName!!)
+// fun Prisoner.personName(): PersonName =
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/DocumentFormatters.kt:81]: Delete commented lines?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/dossier/DossierContextTest.kt [hunk 4]
```diff
@@ -18,28 +20,27 @@ class DossierContextTest {
...
   private val recall = Recall(
     ::RecallId.random(),
-    nomsNumber, ::UserId.random(), OffsetDateTime.now(),
-    OffsetDateTime.now(),
+    nomsNumber, ::UserId.random(), OffsetDateTime.now(), firstName, null, lastName,
     bookingNumber = bookingNumber,
     licenceConditionsBreached = licenceConditionsBreached
   )
-  private val prisoner = Prisoner(firstName = firstName, lastName = lastName)
+  private val prisoner = Prisoner(firstName = firstName.value, lastName = lastName.value)
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/dossier/DossierContextTest.kt:31]: Idea is telling me this as un-used?

**Review by AndyNewton — APPROVED** [review]
Take a quick look at comments please.

**Review by AndyNewton — APPROVED** [review]
🤞

**Pull request merged**